xcode/Subconscious/Shared/Components/AddressBook/AddressBookModel.swift

```diff
@@ -61,25 +166,93 @@ struct AddressBookModel: ModelProtocol {
                 }
             }
             
-            return Update(state: model)
+            let fx: Fx<AddressBookAction> =
+                environment.data.addressBook.listEntries()
+                .map({ follows in
+                    AddressBookAction.populate(follows)
+                })
+                .eraseToAnyPublisher()
             
-        case .follow(did: let did, petname: let petname):
+            return Update(state: model, fx: fx)
+            
+        case .requestFollow:
+            var model = state
+            // Show errors on any untouched fields, hints at why you cannot submit
+            model.followUserForm.did = FormField.update(
+                state: model.followUserForm.did,
+                action: .markAsTouched,
+                environment: FormFieldEnvironment()
+            ).state
+            model.followUserForm.petname = FormField.update(
+                state: model.followUserForm.petname,
+                action: .markAsTouched,
+                environment: FormFieldEnvironment()
+            ).state
+            
+            guard let did = state.followUserForm.did.validated else {
```

Comment:
Comment: I generally try not to reach down more than one level to prevent tight coupling. But I think this is fine here. No change needed.

Reply:
I agree, in a more complex form I think there's value in a `.validated` property (on `followUserForm` in this case) that wraps the entire form payload into a single validated struct. That's often convenient but alsos avoid reaching down two levels.